File: Q05.hs
import Text.Printf (printf)

main :: IO ()
main = do
    let menorIPCA = 1 / 0 -- infinito positivo
        maiorIPCA = -1 / 0 -- infinito negativo
    putStrLn "Digite os valores do IPCA por mês (Digite '*' para encerrar):"
    loop menorIPCA maiorIPCA 0 0 ""

loop :: Double -> Double -> Double -> Int -> String -> IO ()
loop menorIPCA maiorIPCA somaIPCA quantidadeMeses mesMenorIPCA = do
    putStr "Digite o mês e ano (MM/YYYY) seguido do IPCA (utilize vírgula para decimais): "
    entrada <- getLine
    if entrada == "*"
        then do
            let mediaIPCA = somaIPCA / fromIntegral quantidadeMeses
            printf "Menor IPCA: %.2f (%s)\n" menorIPCA mesMenorIPCA
            printf "Maior IPCA: %.2f\n" maiorIPCA
            printf "Média do IPCA: %.2f\n" mediaIPCA
        else do
            let (mesAno, ipca) = readInput entrada
            let menorIPCA' = min menorIPCA ipca
            let maiorIPCA' = max maiorIPCA ipca
            loop menorIPCA' maiorIPCA' (somaIPCA + ipca) (quantidadeMeses + 1) (if ipca < menorIPCA then mesAno else mesMenorIPCA)

readInput :: String -> (String, Double)
readInput entrada = (mesAno, ipca)
    where
        (mesAno, ipcaStr) = break (== ' ') entrada
        ipca = read (tail ipcaStr)

File: Q05.java
import java.util.Scanner;

public class  Q05 {

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        System.out.println("Digite os valores do IPCA por mês (Digite '*' para encerrar):");

        double menorIPCA = Double.MAX_VALUE;
        double maiorIPCA = Double.MIN_VALUE;
        double somaIPCA = 0;
        int quantidadeMeses = 0;
        String mesMenorIPCA = "";
        String mesMaiorIPCA = "";

        while (true) {
            System.out.print("Digite o mês e ano (MM/YYYY) seguido do IPCA (utilize vírgula para decimais): ");
            String entrada = scanner.nextLine().trim();
            if (entrada.equals("*")) {
                break;
            }

            String[] partes = entrada.split(" ");
            String mesAno = partes[0];
            double ipca = Double.parseDouble(partes[1].replace(',', '.'));

            if (ipca < menorIPCA) {
                menorIPCA = ipca;
                mesMenorIPCA = mesAno;
            }

            if (ipca > maiorIPCA) {
                maiorIPCA = ipca;
                mesMaiorIPCA = mesAno;
            }

            somaIPCA += ipca;
            quantidadeMeses++;
        }

        double mediaIPCA = somaIPCA / quantidadeMeses;

        System.out.printf("Menor IPCA: %.2f (%s)\n", menorIPCA, mesMenorIPCA);
        System.out.printf("Maior IPCA: %.2f (%s)\n", maiorIPCA, mesMaiorIPCA);
        System.out.printf("Média do IPCA: %.2f\n", mediaIPCA);
    }
}
